cmd/controller/controller.go

```diff
@@ -225,12 +225,17 @@ func (c *CmdOpts) startController() error {
 	// common factory to get the admin kube client that's needed in many components
 	adminClientFactory := kubernetes.NewAdminClientFactory(c.K0sVars)
 
+	if !c.EnableWorker && stringslice.Contains(c.DisableComponents, constant.KonnectivityServerComponentName) {
+		logrus.Warnf("ignoring disableComponents: %q, because kube workerloads are not enabled on this controller", constant.KonnectivityServerComponentName)
+	}
+	enableKonnectivity := !c.EnableWorker || (!c.SingleNode && !stringslice.Contains(c.DisableComponents, constant.KonnectivityServerComponentName))
```

Comment:
is there any way to simplify this e.g. by changing the logic a bit. Having bunch of negated bools makes this super hard to read, at least for me. :)

Reply:
Sorry for that. How about using `disableKconnectivity` :
```go
disableKconnectivity := c.SingleNode || (c.EnableWorker && stringslice.Contains(c.DisableComponents, constant.KonnectivityServerComponentName))
...
if !disableKconnectivity  {
    ....
}
```

Reply:
that's at least bit easier to read IMO :)

Reply:
I changed the logic a bit and pushed the code changes, hope it's easier to understand.

Reply:
>I changed the logic a bit and pushed the code changes, hope it's easier to understand.

@jnummelin Is this code change OK for you or you want more edits? Please see https://github.com/k0sproject/k0s/commit/1fe7472907064461b2d37ad1e7be89478a05bdaf